### PR TITLE
IPROTO-33 ClassCastException with Array of Enums

### PR DESCRIPTION
**Pull request by anistor**
https://issues.jboss.org/browse/IPROTO-33

### CODE REVIEW

**Comment by anistor** [comment]
@gustavonalle I've taken the minimal stuff needed for Account.currencies from your branch and added the fix in ```ProtoStreamReaderImpl.readCollection```.  The relevant test is ```MarshallingTest.testMarshallAccount```.

File: core/src/test/java/org/infinispan/protostream/MarshallingTest.java [hunk 1]
```diff
@@ -62,6 +62,7 @@ public void testMarshallAccount() throws Exception {
       Account account = new Account();
       account.setId(1);
       account.setDescription("test account");
+      account.setCurrencies(new Account.Currency[]{Account.Currency.BRL});
       Date creationDate = new Date();
```
Comment (gustavonalle) [inline core/src/test/java/org/infinispan/protostream/MarshallingTest.java]: Could you add an assert that the enum value correclty survives marshaling and unmarshalling?

**Comment by gustavonalle** [comment]
Thanks for the ultra-fast fix @anistor, I have just one comment

File: core/src/test/java/org/infinispan/protostream/MarshallingTest.java (hunk 1 above)
Comment (anistor) [inline core/src/test/java/org/infinispan/protostream/MarshallingTest.java]: Done.

**Comment by gustavonalle** [comment]
Merged, thanks @anistor

**Pull request merged**